src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,228 @@
 package core.basesyntax;
 
+import java.util.Map;
+import java.util.Objects;
+
+import static core.basesyntax.MyHashMap.Node.hash;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static int CAPACITY = 16;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = new Node[CAPACITY];
+        threshold = (int) (CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size > threshold) {
+            // resize();
+            //return;
+        }
+        int index = hash(key);
+        Node<K, V> newNode = new Node<>(key, value, null);
+        Node<K, V> node = table[index];
+        //linked list is empty
+        if (node == null) {
+            table[index] = newNode;
+            size++;
+            //linked list isn't empty
+        } else {
+            //оно не заходит сюда
+            //node.key.equals(key)
+            while (node.next != null || checkKey(key, node)) {
+                if (checkKey(key, node)) {
+                    node.value = value;
+                    return;
+                }
+                node = node.next;
+            }
+            //okay if we don't have the same keys -> we should put new pair
+            node.next = newNode;
+            size++;
+        }
     }
 
+
     @Override
     public V getValue(K key) {
+        //maybe we should use next???
+        //should we check that key is present???
+        //if key == null
+        //if linkedList in hashmap isn't empty
+        int index = hash(key);
+        Node<K, V> node = table[index];
+        //it means that key isn't present
+        if (node == null) {
+            return null;
+        }
+
+//        if (checkKey(key, node)) {
+//            return node.value;
+//        }
+
+//        //linked list has one element
+//        if (node.next == null) {
+//            return table[index].value;
+//        }
+
+        while (node != null) {
+            if (checkKey(key, node)) {
+                return node.value;
+            }
+            node = node.next;
+        }
+
+
         return null;
     }
 
+
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        int newCapacity = CAPACITY * 2;
+        Node<K, V>[] newTable = new Node[newCapacity];
+        int index = 0;
+        int sizeOfArr = CAPACITY;
+        CAPACITY = newCapacity;
+
+        for (int i = 0; i < sizeOfArr; i++) {
```

Comment:
The variable `sizeOfArr` is unnecessary. You can directly use `CAPACITY` in the for loop since you have not yet updated the `CAPACITY` with the new capacity. Remember to update the `CAPACITY` after rehashing the nodes.